src/model_theory/basic.lean

```diff
@@ -216,6 +216,19 @@ lemma ext_iff {f g : M ↪[L] N} : f = g ↔ ∀ x, f x = g x :=
 
 lemma injective (f : M ↪[L] N) : function.injective f := f.to_embedding.injective
 
+/-- In an algebraic language, any injective homomorphism is an embedding. -/
+def of_injective [L.is_algebraic] {f : M →[L] N} (hf : function.injective f) : M ↪[L] N :=
+{ inj' := hf,
+  map_rel' := λ n r, (is_algebraic.empty_relations n r).elim,
+  .. f }
+
+@[simp] lemma coe_fn_of_injective [L.is_algebraic] {f : M →[L] N} (hf : function.injective f) :
```

Comment:
I think `@[simps]` in front of the `def` might be able to generate some of these lemmas.